### PR TITLE
add crate for upgrade in plugin

### PR DESCRIPTION
**Pull request by sinhaashish**
This PR creates a new crate for upgrade inside k8s. 
This is for addressing this review comment https://github.com/openebs/mayastor-extensions/pull/114#pullrequestreview-1309085117.

### CODE REVIEW

**Review by niladrih — APPROVED** [review]

File: console-logger/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub mod console;
```
Comment (Abhinandan-Purkait) [inline console-logger/src/lib.rs:1]: Please add doc comments for pub modules.

File: k8s/upgrade/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+pub mod constant;
```
Comment (Abhinandan-Purkait) [inline k8s/upgrade/src/lib.rs:1]: Doc comments needed. Below as well.

File: k8s/upgrade/src/upgrade_resources/mod.rs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pub mod objects;
```
Comment (Abhinandan-Purkait) [inline k8s/upgrade/src/upgrade_resources/mod.rs:1]: Need here as well.

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**